runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/tasks/AbstractDokkaTask.kt

```diff
@@ -232,6 +232,7 @@ abstract class AbstractDokkaTask : DefaultTask() {
 
     init {
         group = JavaBasePlugin.DOCUMENTATION_GROUP
+        super.notCompatibleWithConfigurationCache("Dokka tasks are not yet compatible with the Gradle configuration cache. See https://github.com/Kotlin/dokka/issues/1217")
```

Comment:
Unfortunately, the integration tests are failing :(

```
Caused by: java.lang.NoSuchMethodError: 'void org.gradle.api.DefaultTask.notCompatibleWithConfigurationCache(java.lang.String)
```

The used method (`notCompatibleWithConfigurationCache`) was introduced in Gradle 7.4, whereas we still have to keep compatibility with Gradle 6.9+ (for as long as Kotlin itself supports it).

We probably need to use something like [plugin variants](https://docs.gradle.org/8.3/userguide/implementing_gradle_plugins.html#plugin-with-variants) to fix this properly, but there's little point in investing time into the current Gradle plugin - it will be completely re-written anyway.

However, it would still be nice to have this change for the time being. I think we can add a dirty hack - check with reflection that this method exists, and only then call it. What do you think?

Something along the lines of

```kotlin
val containsNotCompatibleWithConfigurationCache = this::class.memberFunctions
    .any { it.name == "notCompatibleWithConfigurationCache" && it.parameters.firstOrNull()?.name == "reason" }

if (containsNotCompatibleWithConfigurationCache) {
    super.notCompatibleWithConfigurationCache("Dokka tasks are not yet compatible with the Gradle configuration cache. See https://github.com/Kotlin/dokka/issues/1217")
}
```

Reply:
Yes I think using introspection is a fair workaround :) Feel free to amend the PR (or I can do it if you prefer). 👍

Reply:
I don't want to rob you of the valuable contribution :sweat_smile: If you have the time and the desire, I'd be happy if you did it :) I also didn't test the suggestion - I mean, it should work if this function is returned in `memberFunctions`, but who knows

Reply:
Thanks! I've just pushed the change. It seems to work on my side but let's see if the CI agrees :)